src/Symfony/Component/HttpKernel/MicroKernel.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel;
+
+use Symfony\Component\Config\Loader\Loader;
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Loader\ContainerBuilderAwareLoader;
+
+/**
+ * A Kernel that allows you to configure services.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+abstract class MicroKernel extends Kernel
+{
+    /**
+     * Configure dependency injection extensions that have been added to the container.
+     *
+     * $c->loadFromExtension('framework', array(
+     *     'secret' => '%secret%'
+     * ));
+     *
+     * @param ContainerBuilder $c
+     * @param LoaderInterface  $loader
+     */
+    abstract protected function configureExtensions(ContainerBuilder $c, LoaderInterface $loader);
+
+    /**
+     * Add any service definitions to your container.
+     *
+     * @param ContainerBuilder $c
+     * @param LoaderInterface  $loader
+     */
+    abstract protected function configureServices(ContainerBuilder $c, LoaderInterface $loader);
+
+    /**
+     * Applies the bundle configuration and calls configureServices() for continued building.
+     *
+     * @param LoaderInterface $loader
+     */
+    public function registerContainerConfiguration(LoaderInterface $loader)
+    {
+        if (!$loader instanceof ContainerBuilderAwareLoader) {
```

Comment:
This should maybe be an interface, but the class is so simple and we just use it to fetch the ContainerBuilder and *real* Loader out, so I don't know if there's a real-use case to pass other concrete implementation in.